fah_xchem/analysis/report.py

```diff
@@ -132,6 +137,142 @@ def RenderData(image, mol, tags):
         table.DrawText(cell, value)
 
 
+def generate_fragalysis(
+    series: CompoundSeriesAnalysis,
+    fragalysis_config: FragalysisConfig,
+    results_path: str,
+) -> None:
+
+    """
+    Generate input and upload to fragalysis from ligand_filename and proteins_filename​
+
+    Fragalysis spec:https://discuss.postera.ai/t/providing-computed-poses-for-others-to-look-at/1155/8?u=johnchodera​
+
+    Parameters
+    ----------
+    series : CompoundSeriesAnalysis
+        Analysis results
+    fragalysis_config : FragalysisConfig
+        Fragalysis input paramters
+    results_path : str
+        The path to the results
+    """
+
+    import os
+    from openeye import oechem
+    from rich.progress import track
+
+    ref_mols = fragalysis_config.ref_mols  # e.g. x12073
+    ref_pdb = fragalysis_config.ref_pdb  # e.g. x12073
+
+    # set paths
+    ligands_path = os.path.join(results_path, fragalysis_config.ligands_filename)
+    proteins_path = os.path.join(results_path, fragalysis_config.proteins_filename)
+    fa_ligands_path = os.path.join(
+        results_path, fragalysis_config.fragalysis_sdf_filename
+    )
+
+    # copy sprint generated sdf to new name for fragalysis input
+    from shutil import copyfile
+
+    copyfile(ligands_path, fa_ligands_path)
+
+    # Read ligand poses
+    molecules = []
+
+    with oechem.oemolistream(ligands_path) as ifs:
+        oemol = oechem.OEGraphMol()
+        while oechem.OEReadMolecule(ifs, oemol):
+            molecules.append(oemol.CreateCopy())
+    print(f"{len(molecules)} ligands read")
+
+    descriptions = {
+        "DDG (kcal/mol)": "Relative computed free energy difference",
```

Comment:
We might eventually update this to DG and put the absolute free energy and predicted pIC50, but this is great for now.